app/controllers/grade_entry_forms_controller.rb

```diff
@@ -405,6 +405,15 @@ def csv_upload
         flash[:error] = "You did not upload a .csv file."
       end
     end
+    
+    # Replace non-UNIX line endings with standard CR+LF style
+    if (reader = File.read(params[:upload][:grades_file].tempfile,
+                           mode: 'rb', encoding: 'UTF-8'))
+      replaced_newlines = reader.gsub!(/\r\n?|\n/, "\r\n")
+      File.open(params[:upload][:grades_file].tempfile, "wb") do |f|
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.